### PR TITLE
Add desktop passkey CLI POC (browser-first + CTAP fallback) and strategy docs

### PR DESCRIPTION
**Pull request by szijpeter**
## Summary
- add new non-published sample module `samples:passkey-cli` (JVM + application plugin)
- implement CLI commands: `doctor`, `register`, `authenticate`
- reuse existing backend/network contracts via `KtorPasskeyServerClient`
- add sample-local authenticator abstraction (`AuthenticatorAdapter`) with:
  - `BrowserHandoffAdapter` (default `browser` mode for platform passkeys)
  - `PythonFido2Adapter` (optional `ctap` mode for native security-key flow)
- add Python bridge script for CTAP-based authenticator invocation (`python-fido2`) and sample-local venv bootstrap
- add backend browser bridge route/page (`/webauthn/cli/browser`) for CLI browser handoff
- add parser/doctor UX improvements:
  - `--authenticator <browser|ctap>`
  - endpoint/rp/origin defaults derived from endpoint
  - auto-default to `local.properties` `WEBAUTHN_DEMO_*` values when present
- add tests for parser/routing, orchestration paths, browser handoff, and JSON/process boundary behavior
- add desktop/CLI strategy document: `docs/DESKTOP_CLI_STRATEGY.md`
- update docs for local smoke setup and HTTPS/ngrok workflow
- keep sample-only scope (no published API changes)

## Scope
- sample-only POC and docs
- no published module API changes
- no BCV baseline updates

## Validation
- `./gradlew :samples:passkey-cli:test :samples:passkey-cli:detekt --stacktrace` ✅
- `./gradlew :samples:backend-ktor:test :samples:backend-ktor:detekt --stacktrace` ✅
- `tools/agent/quality-gate.sh --mode fast --scope changed --block false` ✅
- `tools/agent/quality-gate.sh --mode strict --scope changed --block false` ✅

## Notes
- existing unrelated unstaged change in `samples/compose-passkey/.../Header.kt` remains out of scope and was not included in commits from this thread.
- auto-closing the browser bridge tab was attempted and reverted due to unreliable browser behavior (especially Chrome) for externally opened tabs.

### CODE REVIEW

**Comment by szijpeter** [comment]
@codex review

File: samples/passkey-cli/src/main/kotlin/dev/webauthn/samples/passkeycli/DoctorCommandRunner.kt [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+            return true
+        }
+        stderr.appendLine("Bridge script check failed: '${bridgePath.toAbsolutePath()}' does not exist.")
+        return false
+    }
+
+    private suspend fun checkPython(pythonBinary: String): Boolean {
+        val pythonVersion = commandExecutor.execute(
+            command = listOf(pythonBinary, "--version"),
+        )
```
[the remark this answers is not in the document]
Comment (szijpeter) [reply samples/passkey-cli/src/main/kotlin/dev/webauthn/samples/passkeycli/DoctorCommandRunner.kt]: Implemented in 0374c60. `doctor` now catches process-launch failures in `checkPython`, `checkPythonFido2`, and `probeDevices`, reports descriptive stderr messages, and returns normal FAIL output instead of aborting.

**Comment by szijpeter** [comment]
Addressed the actionable review feedback in commit `0374c60`.

Included fixes:
- Python bridge now emits structured failure envelopes on stdout (`fido2_bridge.py`).
- CLI parse/usage failures now return a distinct exit code (`EXIT_PARSE_USAGE = 64`) instead of colliding with ceremony option/start failures.
- `doctor` no longer aborts on process launch failures; each check reports a descriptive failure/warning and continues.
- Rejected server results now use a null-safe fallback message (`no reason provided`) instead of printing `null`.

Also added regression tests for these paths (`CliApplicationTest`, `DoctorCommandRunnerTest`, and additional ceremony/adapter tests).

File: samples/passkey-cli/src/main/kotlin/dev/webauthn/samples/passkeycli/PasskeyCeremonyRunner.kt [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+    }
+
+    private suspend fun resolveRegisterOptions(
+        payload: RegistrationStartPayload,
+    ): PublicKeyCredentialCreationOptions? {
+        val result = runCatching { serverClient.getRegisterOptions(payload) }
+            .getOrElse { error ->
+                stderr.appendLine("Failed to fetch registration options: ${error.message}")
+                return null
+            }
```
[the remark this answers is not in the document]
Comment (szijpeter) [reply samples/passkey-cli/src/main/kotlin/dev/webauthn/samples/passkeycli/PasskeyCeremonyRunner.kt]: Addressed in `8e24ef6`.

I now rethrow `CancellationException` in all six `runCatching { ... }.getOrElse { ... }` branches in `PasskeyCeremonyRunner`, and added a regression test (`register_whenStartIsCancelled_rethrowsCancellationException`) to verify cancellation propagates instead of being swallowed.

**Comment by szijpeter** [comment]
Pushed `8e24ef6` with self-contained venv improvements for `samples/passkey-cli` and addressed the remaining actionable review item.

Included in this commit:
- auto-detect local `samples/passkey-cli/.venv/bin/python` (fallback to `python3`) in `CliParser`
- new bootstrap flow: `:samples:passkey-cli:bootstrapVenv` + `scripts/bootstrap_venv.sh` + `requirements.txt`
- README updates for single-command sample-local setup
- fixed cancellation handling in `PasskeyCeremonyRunner` by rethrowing `CancellationException` in all error branches
- regression tests for venv auto-detection and cancellation propagation

Validation:
- `./gradlew :samples:passkey-cli:test :samples:passkey-cli:detekt --stacktrace` ✅
- `tools/agent/quality-gate.sh --mode fast --scope changed --block false` ✅
- `tools/agent/quality-gate.sh --mode strict --scope changed --block false` ✅

File: samples/passkey-cli/src/test/kotlin/dev/webauthn/samples/passkeycli/CliParserTest.kt [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+                "alice",
+            ),
+        )
+
+        assertTrue(invocation is CliInvocation.Register)
+        assertEquals("http://127.0.0.1:8080", invocation.common.endpointBase)
+        assertEquals("localhost", invocation.common.rpId)
+        assertEquals("https://localhost", invocation.common.origin)
+        assertEquals("python3", invocation.common.pythonBinary)
+        assertEquals("alice", invocation.userDisplayName)
```
[the remark this answers is not in the document]
Comment (szijpeter) [reply samples/passkey-cli/src/test/kotlin/dev/webauthn/samples/passkeycli/CliParserTest.kt]: Addressed in `46b708f`.

I switched parser tests to isolated temp directories (instead of `/tmp/webauthn`) so default-python assertions are deterministic regardless of host `.venv` state.

**Comment by szijpeter** [comment]
Follow-up for latest review feedback: pushed `46b708f`.

Addressed:
- fixed `CliParserTest` flakiness by using isolated temp `cwd` values instead of hardcoded `/tmp/webauthn`
- refactored duplicated upward traversal in `CliParser` via a shared private helper `findUpwards(...)`

Validation:
- `./gradlew :samples:passkey-cli:test :samples:passkey-cli:detekt --stacktrace` ✅
- `tools/agent/quality-gate.sh --mode fast --scope changed --block false` ✅
- `tools/agent/quality-gate.sh --mode strict --scope changed --block false` ✅

**Comment by szijpeter** [comment]
Checked latest review feedback and pushed follow-up `f5b33ec`.

Addressed:
- `bootstrapVenv` now declares Gradle inputs/outputs (`scripts/bootstrap_venv.sh`, `requirements.txt`, `.venv`) for better up-to-date behavior.
- `PasskeyCeremonyRunner` now uses null-safe throwable rendering (`displayMessage()`) in all failure log branches to avoid `...: null` output.

Validation:
- `./gradlew :samples:passkey-cli:test :samples:passkey-cli:detekt --stacktrace` ✅
- `tools/agent/quality-gate.sh --mode fast --scope changed --block false` ✅
- `tools/agent/quality-gate.sh --mode strict --scope changed --block false` ✅

**Comment by szijpeter** [comment]
Pushed follow-up `994fe1f` to implement the browser-orchestrated platform-passkey path for desktop/CLI while keeping `ctap` as fallback.

Included:
- new `browser` authenticator mode (default) and browser handoff adapter in `samples/passkey-cli`
- backend-hosted bridge page route at `/webauthn/cli/browser`
- doctor checks split by mode (`browser` vs `ctap`)
- docs and tests updated for browser-first behavior

Validation run locally:
- `./gradlew :samples:passkey-cli:test :samples:backend-ktor:test --stacktrace` ✅
- `tools/agent/quality-gate.sh --mode fast --scope changed --block false` ✅
- `tools/agent/quality-gate.sh --mode strict --scope changed --block false` ✅

Checked review thread state: current PR review threads are resolved.

**Comment by szijpeter** [comment]
Follow-up pushed: `f98e9ab`.

What changed:
- `samples/passkey-cli` now auto-reads `WEBAUTHN_DEMO_ENDPOINT`, `WEBAUTHN_DEMO_RP_ID`, and `WEBAUTHN_DEMO_ORIGIN` from nearest `local.properties` as defaults.
- Explicit CLI flags still win.
- If only `--endpoint` is overridden, `rpId`/`origin` now derive from that endpoint (avoids stale local-properties mismatch).
- Added parser tests for local-properties defaults + endpoint override behavior.
- Updated sample README and CLI usage text to document the new default behavior.

Validation:
- `./gradlew :samples:passkey-cli:test :samples:passkey-cli:detekt --stacktrace` ✅

**Comment by szijpeter** [comment]
Checked latest PR feedback and pushed follow-up `cde8de4`.

Addressed:
- Fixed browser-launch fallback flow in `BrowserHandoffAdapter`: when launcher returns `false` or throws, we now keep callback server alive, print manual-open guidance, and continue waiting for callback (with proper `finally` cleanup).
- Added `BrowserHandoffAdapterTest` coverage for launcher return-`false` and launcher-throws branches.
- Added an explanatory comment for sample `rpName` reuse (`rpName = rpId`) in `PasskeyCeremonyRunner`.
- Added freshness qualifier + references in `docs/DESKTOP_CLI_STRATEGY.md` ecosystem-positioning bullets.

Validation:
- `./gradlew :samples:passkey-cli:test :samples:passkey-cli:detekt --stacktrace` ✅
- `tools/agent/quality-gate.sh --mode fast --scope changed --block false` ✅
